src/modules/functionality.js

```diff
@@ -0,0 +1,141 @@
+  taskObject = {
+    description: input.value,
+    completed: false,
+    index: taskStorage.length + 1,
+  };
+  if (taskObject.description !== '') {
+    taskStorage.push(taskObject);
+    saveTodo();
+  }
+};
+
+export default function addEditRemoveTask(task) {
+  // function for adding new task to the list
+  const taskContainer = document.createElement('div');
+  taskContainer.classList.add('task');
+  const taskContentContainer = document.createElement('div');
+  taskContentContainer.classList.add('content');
+  taskContainer.appendChild(taskContentContainer);
+  const taskCheckboxContainer = document.createElement('input');
+  taskCheckboxContainer.classList.add('checkbox');
+  taskCheckboxContainer.type = 'checkbox';
+  const taskInputContainer = document.createElement('input');
+  taskInputContainer.classList.add('text');
+  taskInputContainer.type = 'text';
+  taskInputContainer.value = task.description;
+  taskInputContainer.setAttribute('readonly', 'readonly');
+  taskContentContainer.appendChild(taskCheckboxContainer);
+  taskContentContainer.appendChild(taskInputContainer);
+
+  // function for editing and removing task from the list
+  const taskActionsContainer = document.createElement('div');
+  taskActionsContainer.classList.add('actions');
+  const taskEditButton = document.createElement('button');
+  taskEditButton.classList.add('edit');
+  taskEditButton.innerText = 'Edit';
+  const taskDeleteButton = document.createElement('button');
+  taskDeleteButton.classList.add('delete');
+  taskDeleteButton.innerText = 'Delete';
+  taskActionsContainer.appendChild(taskEditButton);
+  taskActionsContainer.appendChild(taskDeleteButton);
+  taskContainer.appendChild(taskActionsContainer);
+  listContainer.appendChild(taskContainer);
+  input.value = '';
+
+  taskEditButton.addEventListener('click', () => {
+    if (taskEditButton.innerText.toLowerCase() === 'edit') {
+      taskEditButton.innerText = 'Save';
+      taskInputContainer.removeAttribute('readonly');
+      taskInputContainer.focus();
+    } else {
+      taskEditButton.innerText = 'Edit';
+      taskInputContainer.setAttribute('readonly', 'readonly');
+    }
+  });
+
+  // function for removing task from the list
+  taskDeleteButton.addEventListener('click', () => {
+    listContainer.removeChild(taskContainer);
+  });
+
+  // Remove the todo item from the list of todo items in local storage
+  taskDeleteButton.addEventListener('click', () => {
+    const taskIndex = taskStorage.indexOf(task);
+    taskStorage.splice(taskIndex, 1);
+    saveTodo();
+  });
+
+  // delete all tasks from the list
+  deleteAllList();
+
+  // Remove all todo items from the list in local storage
+  removeAllTodo();
+
+  // function for updating items object's value for completed
+  taskCheckboxContainer.addEventListener('change', () => {
+    if (taskCheckboxContainer.checked === true) {
+      task.completed = true;
+    } else {
+      task.completed = false;
+    }
+    saveTodo();
+  });
+
```

Comment:
![Capture1](https://user-images.githubusercontent.com/93039359/207962752-d9dd5665-3d3d-4e8e-b962-c487e82eb76e.PNG)

- [ ] You are supposed to add a feature which is if you check some of your todos then there should be a button which will remove all the checked todos. You have a button for deleting all todos but it would be nice if you add the functionality of clearing all checked todos 😊

Reply:
Issues fixed. Thank you for your reviews